### PR TITLE
[Smart Wallet] Use bundler eth_getUserOperationReceipt instead of rpc eth_getLogs

### PR DESCRIPTION
**Pull request by 0xFirekeeper**
Reduce inconsistencies across smart wallet production flows and unit test flows.

---

## PR-Codex overview
This PR updates the code related to gas fees, smart wallets, Unity SDK version, test timeouts, and RPC client methods.

### Detailed summary
- Updated handling of gas station errors in `gas-price.ts`
- Updated dependencies in `fifty-chefs-learn.md`
- Updated Unity SDK version in `thirdweb-bridge.ts`
- Adjusted test timeout in `package.json`
- Added new RPC client method in `http-rpc-client.ts`
- Improved error handling in `base-api.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/sdk/src/evm/common/gas-price.ts [hunk 1]
```diff
@@ -188,8 +188,8 @@ export async function getPolygonGasPriorityFee(
       const fixedFee = parseFloat(priorityFee).toFixed(9);
       return utils.parseUnits(fixedFee, "gwei");
     }
-  } catch (e) {
-    console.error("failed to fetch gas", e);
+  } catch {
+    // if the gas station is down, return the default gas fee
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/common/gas-price.ts:192]: Worried about this error. If they changed the response maybe we just need to update it ?

File: packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts [hunk 2]
```diff
@@ -342,28 +342,31 @@ export abstract class BaseAccountAPI {
...
-        this.entryPointView.filters.UserOperationEvent(userOpHash),
-      );
-      if (events[0]) {
-        return events[0].transactionHash;
+      const userOpReceipt =
+        await httpRpcClient.getUserOperationReceipt(userOpHash);
+      if (userOpReceipt) {
+        // avoid desync with current provider state
+        return await this.provider.waitForTransaction(
+          userOpReceipt.receipt.transactionHash,
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts:364]: Interesting that it returns the hash before the provider knows about it, guess nodes might not be in full sync?

File: packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-provider.ts [hunk 3]
```diff
@@ -134,7 +123,13 @@ export class ERC4337EthersProvider extends providers.BaseProvider {
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         confirmations?: number,
       ): Promise<providers.TransactionReceipt> => {
-        const transactionReceipt = await waitForUserOp();
+        const transactionReceipt =
+          await new Promise<providers.TransactionReceipt>((resolve, reject) => {
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-provider.ts:127]: Do you need to wrap in a promise here?

File: packages/sdk/src/evm/common/gas-price.ts (hunk 1 above)
Comment (0xFirekeeper) [reply packages/sdk/src/evm/common/gas-price.ts:192]: Most likely 403s (at least those happen in Unity sometimes) but yeah would need to debug 🤔

File: packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts (hunk 2 above)
Comment (0xFirekeeper) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts:364]: Yep! At least one block off usually.

File: packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-provider.ts (hunk 3 above)
Comment (0xFirekeeper) [reply packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-provider.ts:127]: Without it I was getting some issues, though that was before the desync fix, will test without wrapping and get back.

File: packages/sdk/src/evm/common/gas-price.ts (hunk 1 above)
Comment (0xFirekeeper) [reply packages/sdk/src/evm/common/gas-price.ts:192]: seems it was just down, works fine now, I think this is the right behavior if that's the case

**Pull request merged**